internal/event/dispatch_service.go

```diff
@@ -145,12 +149,13 @@ func (s *DispatchService) Unsubscribe(eventName string, subscriptionID int64) {
 // If this method returns true, it is guaranteed that the dispatch service close wait for all events to be handled.
 // Returns false if Dispatch Service is already closed.
 func (s *DispatchService) Publish(event Event) bool {
-	if atomic.LoadInt32(&s.state) == stopped {
+	s.subscriptionsMu.RLock()
+	defer s.subscriptionsMu.RUnlock()
+
+	if s.state == stopped {
 		return false
 	}
```

Comment:
Could you change this to:
```
	s.subscriptionsMu.RLock()
	defer s.subscriptionsMu.RUnlock()
	if atomic.LoadInt32(&s.state) == stopped {
		return false
	}

```

Reply:
`stopped` serves a different purpose than `subscriptionsMu`, it is there to make sure the "stop" logic for the dispatch service executes only once. It's being accessed atomically in order to protect concurrent access to it. `subscrptionsMu` protects something else.